scheduler/src/main/java/io/camunda/zeebe/scheduler/ActorJob.java

```diff
@@ -31,15 +31,18 @@ public final class ActorJob {
   private boolean isDoneCalled;
   private ActorFuture resultFuture;
   private ActorSubscription subscription;
+  private long scheduledAt = -1;
 
   public void onJobAddedToTask(final ActorTask task) {
+    scheduledAt = System.nanoTime();
```

Comment:
:question: :wrench: Any reason why we not use the Timer API from the prometheus metric client? I can remember that we decided once to always use that instead of self calculating the values.

Reply:
I don't think we can do that because we want to observe specific durations for timer subscriptions and such. See `observeSchedulingLatency`. `Histogram.Timer` doesn't allow that, it only offers a `observeDuration()` method that does the measurement itself.

Reply:
What about the method `startTimer` and then calling close?

Reply:
But we don't want to start the timer for jobs that have a timer subscription, otherwise we would mostly measure the timer duration (often in the minutes, for snapshots and such). In that case we want to measure the time between timer expiration and job being executed. Since the job doesn't directly know when the timer expires (there's no callback or anything) it can't start the timer at that point and so we need to calculate the duration ourselves.

Reply:
Not sure whether I could follow here :sweat_smile: I thought we can store the reference to the timer and then close it if necessary but I'm also ok with this.